ui/packages/catalog-ui-search/src/main/webapp/react-component/container/theme-settings/theme-settings.tsx

```diff
@@ -0,0 +1,353 @@
+    this.state = {
+      fontSizeModel: new Property({
+        isEditing: true,
+        label: 'Zoom Percentage',
+        value: [themeScale],
+        min: 62,
+        max: 200,
+        units: '%',
+        type: 'RANGE',
+      }),
+      spacingModeModel: new Property({
+        isEditing: true,
+        enum: [
+          {
+            label: 'Comfortable',
+            value: 'comfortable',
+          },
+          {
+            label: 'Cozy',
+            value: 'cozy',
+          },
+          {
+            label: 'Compact',
+            value: 'compact',
+          },
+        ],
+        value: [getSpacingMode()],
+        id: 'Spacing',
+      }),
+      animationModel: new Property({
+        isEditing: true,
+        label: 'Animation',
+        value: [getAnimationMode()],
+        enum: [
+          {
+            label: 'On',
+            value: true,
+          },
+          {
+            label: 'Off',
+            value: false,
+          },
+        ],
+        id: 'Animation',
+      }),
+      hoverPreviewModel: new Property({
+        isEditing: true,
+        label: 'Preview Full Image on Hover',
+        value: [user.getHoverPreview()],
+        enum: [
+          {
+            label: 'On',
+            value: true,
+          },
+          {
+            label: 'Off',
+            value: false,
+          },
+        ],
+        id: 'Full Image on Hover',
+      }),
+      themeModel: new Property({
+        isEditing: true,
+        enum: [
+          {
+            label: 'Dark',
+            value: 'dark',
+          },
+          {
+            label: 'Light',
+            value: 'light',
+          },
+          {
+            label: 'Sea',
+            value: 'sea',
+          },
+          {
+            label: 'Custom',
+            value: 'custom',
+          },
+        ],
+        value: [getTheme()],
+        id: 'Theme',
+      }),
+      customToggle: this.isCustomSet(getTheme()),
+      customColors: getCustomColors(),
+    }
+
+    this.state.fontSizeModel.on('change:value', () => {
+      saveFontChanges(this.state.fontSizeModel.getValue()[0])
+    })
+    this.state.spacingModeModel.on('change:value', () => {
+      saveSpacingChanges(this.state.spacingModeModel.getValue()[0])
+    })
+    this.state.animationModel.on('change:value', () => {
+      saveAnimationChanges(this.state.animationModel.getValue()[0])
+    })
+    this.state.hoverPreviewModel.on('change:value', () => {
+      saveHoverPreviewChanges(this.state.hoverPreviewModel.getValue()[0])
+    })
+    this.state.themeModel.on('change:value', () => {
+      let themeValue = this.state.themeModel.getValue()[0]
+      saveThemeChanges(themeValue)
+      this.setState({
+        customToggle: this.isCustomSet(themeValue),
+      })
+    })
+  }
+  isCustomSet(themeValue: string) {
+    return themeValue === 'custom' ? true : false
+  }
+  render() {
+    return (
+      <Root {...this.props}>
+        <div className="theme-font-size">
```

Comment:
Can all these `div`s with classNames be removed - they're not being used for any style targeting, and I think the MarionetteRegionContainer can live on it's own